ssl/statem/extensions_srvr.c

```diff
@@ -283,7 +283,13 @@ int tls_parse_ctos_sig_algs_cert(SSL_CONNECTION *s, PACKET *pkt,
         return 0;
     }
 
-    if (!s->hit && !tls1_save_sigalgs(s, &supported_sig_algs, 1)) {
+    /*
+     * We use this routine on both clients and servers, and when clients
```

Comment:
Interesting. Although unrelated to this PR, it seems that given this comment this function is misnamed. The "ctos" stands for "client to server" - but from the comment this is clearly also used in the "stoc" context as well.

Reply:
Yeah, that was done in 51c7d3e824612a9c71bd987862a00140eb4b0711 for expediency -- allowing sigalgs in TLS 1.3 CertificateRequest was effectively a one-line change by not renaming the function.